pkg/apis/application/v1alpha1/types.go

```diff
@@ -285,6 +286,33 @@ type RepositoryList struct {
 	Items           []Repository `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
+func (app *Application) getFinalizerIndex(name string) int {
+	for i, finalizer := range app.Finalizers {
+		if finalizer == name {
+			return i
+		}
+	}
+	return -1
+}
+
+// NeedPruneResources indicates if resources finalizer is set and controller should delete app resources before deleting app
+func (app *Application) NeedPruneResources() bool {
```

Comment:
I think prune may be incorrect terminology.  Pruning implies something will be left over (e.g. your have resources that is no longer managed in git and you want to delete those). But this field indicates whether or not you should delete all the dependant resources.

How about:
app.CascadeDelete()
app.OrphanDependants()